src/database.ts

```diff
@@ -3242,42 +3243,38 @@ class Database extends common.GrpcServiceObject {
         }
 
         const release = () => {
-          span.end();
           this.pool_.release(session!);
         };
 
         const runner = new TransactionRunner(
           session!,
           transaction!,
-          (err, resp) => {
+          async (err, resp) => {
             if (err) {
               setSpanError(span, err!);
```

Comment:
```suggestion
              setSpanError(span, err);
```

we already are using if condition